src/bsoncxx/view_or_value.hpp

```diff
@@ -123,6 +123,30 @@ class view_or_value {
     View _view;
 };
 
+///
+/// Equality operators for comparison with plain Views.
+///
+
+template <typename View, typename Value>
+BSONCXX_INLINE bool operator==(view_or_value<View, Value> lhs, View rhs) {
```

Comment:
Given that view_or_value may transfer ownership, I'm very certain you want to say `const view_or_value<View, Value>&` here. For the `View` type argument, I don't think it matters.